spec/features/user_show_spec.rb

```diff
@@ -0,0 +1,136 @@
+    sleep(1)
+    expect(page).to have_css("img[src='#{@first_user.photo}']")
+    visit users_path
+    sleep(1)
+    click_link(@second_user.name)
+    sleep(1)
+    expect(page).to have_css("img[src='#{@second_user.photo}']")
+    visit users_path
+    sleep(1)
+    click_link(@third_user.name)
+    sleep(1)
+    expect(page).to have_css("img[src='#{@third_user.photo}']")
+  end
+
+  it "shows the user's username" do
+    visit users_path
+    sleep(1)
+    click_link(@first_user.name)
+    sleep(1)
+    expect(page).to have_content('Tom')
+    visit users_path
+    sleep(1)
+    click_link(@second_user.name)
+    sleep(1)
+    expect(page).to have_content('Lilly')
+    visit users_path
+    sleep(1)
+    click_link(@third_user.name)
+    sleep(1)
+    expect(page).to have_content('Alan')
+  end
+
+  it 'shows the number of posts the user has written' do
+    visit users_path
+    sleep(1)
+    click_link(@first_user.name)
+    sleep(1)
+    expect(page).to have_content('Number of posts: 4')
+    visit users_path
+    sleep(1)
+    click_link(@second_user.name)
+    sleep(1)
+    expect(page).to have_content('Number of posts: 2')
+    visit users_path
+    sleep(1)
+    click_link(@third_user.name)
+    sleep(1)
+    expect(page).to have_content('Number of posts: 0')
+  end
+
+  it "shows the user's bio" do
+    visit users_path
+    sleep(1)
+    click_link(@first_user.name)
+    sleep(1)
+    expect(page).to have_content('Teacher from Mexico.')
+    visit users_path
+    sleep(1)
+    click_link(@second_user.name)
+    sleep(1)
+    expect(page).to have_content('Teacher from Poland.')
+    visit users_path
+    sleep(1)
+    click_link(@third_user.name)
+    sleep(1)
+    expect(page).to have_content('Singer from Mexico.')
+  end
+
+  it "shows the user's latest 3 posts" do
+    visit users_path
+    sleep(1)
+    click_link(@first_user.name)
+    sleep(1)
+    expect(page).to have_content('Post4')
+    expect(page).to have_content('Post3')
+    expect(page).to have_content('Post2')
+  end
+
+  it "shows a button that lets me view all of a user's posts" do
+    visit users_path
+    sleep(1)
+    click_link(@first_user.name)
+    sleep(1)
+    expect(page).to have_link('See all posts', href: "/users/#{@first_user.id}/posts?page=1")
+  end
+
+  context "When I click a user's post" do
+    it "redirects me to that post's show page" do
+      visit users_path
+      sleep(1)
+      click_link(@first_user.name)
+      sleep(1)
+      click_link('Post4')
+      sleep(1)
+      expect(page).to have_current_path(user_post_path(@first_user, @latest_post))
+    end
+  end
+
+  context 'When I click to see all posts' do
+    it "redirects me to the user's post's index page" do
+      visit users_path
+      sleep(1)
+      click_link(@first_user.name)
+      sleep(1)
+      click_link('See all posts')
+      sleep(1)
+      expect(page).to have_current_path("/users/#{@first_user.id}/posts?page=1")
+    end
+  end
+end
```

Comment:
**README.md** file:

- [ ] To make your README looks more professional and as required the used template, kindly ensure that you update the `## Author` section by adding all authors contributors of the project